tracing-attributes/tests/async_fn.rs

```diff
@@ -371,3 +371,119 @@ fn out_of_scope_fields() {
 
     handle.assert_finished();
 }
+
+#[test]
+fn manual_impl_future() {
+    #[instrument]
+    fn manual_impl_future() -> impl Future<Output = ()> {
+        async move {
+            tracing::trace!(poll = true);
+        }
+    }
+    #[instrument]
+    fn manual_box_pin() -> Pin<Box<dyn Future<Output = ()>>> {
+        Box::pin(async move {
+            tracing::trace!(poll = true);
+        })
+    }
```

Comment:
nit, take it or leave it: personally, i think it would be nice if the boxed and non-boxed versions were in separate tests, so that we can always test both cases even if one fails, and know which one is broken. for example, my guess is that the `Box::pin` version currently _does_ work on `master`, while the `impl Future` version doesn't; it would be nice if the tests indicated that.